src/backend/mpi/mpi.cpp

```diff
@@ -180,6 +180,9 @@ namespace argo {
 			return _is_cached(reinterpret_cast<std::size_t>(addr));
 		}
 
+		std::size_t block_size() {
+			return data_distribution::is_cyclic_policy() ? env::allocation_block_size()*page_size : CACHELINE*page_size;
```

Comment:
I am still wondering why this is not instead `data_distribution::block_size` instead of the `is_cyclic_policy()` check here. Is this re-inventing virtual functions, or is there some other reason to use a check here? Is this only about the `CACHELINE` constant? No change requested, just wondering.

Reply:
`data_distribution::block_size` is initialized to `env::block_size()` in every case, which defaults to 16. This is not true for the _first-touch_ or _naive_ page placement policies, which do not have any defined block sizes. The requirement for this function is to return the minimum block size that is contiguous in memory.

- For _first-touch_, that is always going to be one block. (debatable whether this should be `page_size` or `CACHELINE*page_size`, since first-touch does not take `CACHELINE` into account)
- For the naïve page placement policy, this is either `CACHELINE*block_size` or `backend::global_size()/argo::number_of_nodes`.

Reply:
`data_distribution::policy_block_size()` now returns the appropriate block size for the chosen allocation policy. The interface remains the same, but the backend is not directly involved.